db-tests/src/test/java/gov/nasa/jpl/aerie/database/AnchorTests.java

```diff
@@ -192,11 +192,11 @@ private Activity getActivity(final int planId, final int activityId) throws SQLE
         WHERE id = %d
         AND plan_id = %d;
       """.formatted(activityId, planId));
-      res.first();
+      res.next();
```

Comment:
Good question - according to the JDBC spec, by default ResultSets are "forward only", meaning that they support only the `next` operation. 

https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html

Here's the place in the new driver that sets that default:

https://github.com/pgjdbc/pgjdbc/blob/3c3e5b62349c42d9d3dfe1fe0102776d4de8c353/pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java#L515-L519

And here's where it throws an exception if you try to scroll a non-scrollable result set:

https://github.com/pgjdbc/pgjdbc/blob/3c3e5b62349c42d9d3dfe1fe0102776d4de8c353/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java#L312-L319

So, the fact that impossibl's pgjdbc-ng allows us to call `first`, even though the statement's type is "forward only", is a deviation from the spec.

If we want to be able to call `first`, we can specify that when creating the statement, it would look something like this:

```java
connection.createStatement(
        ResultSet.TYPE_SCROLL_SENSITIVE,
        ResultSet.CONCUR_READ_ONLY);
```

My approach in this PR was to update our code to work with the default non-scrollable result sets

Reply:
ooooh, good find